src/selectors/adaptJavaScriptValueToXPathValue.js

```diff
@@ -1,39 +1,50 @@
 import Sequence from './dataTypes/Sequence';
 import createAtomicValue from './dataTypes/createAtomicValue';
 import NodeValue from './dataTypes/NodeValue';
+import ArrayValue from './dataTypes/ArrayValue';
+import MapValue from './dataTypes/MapValue';
 
 function adaptItemToXPathValue (value) {
```

Comment:
Also, none of these branches actually returns more than a singleton sequence. Why not leave the Sequence constructor call where it was?